packages/nonempty-list/src/index.ts

```diff
@@ -75,9 +75,7 @@ export class NonEmptyList<T> implements Iterable<T> {
   };
 
   public join = (separator?: string): string => {
-    const first =
-      typeof separator === 'undefined' ? String(this.first) : String(this.first) + separator;
-    return first + this.rest.join(separator);
+    return this.toArray().join(separator);
```

Comment:
The tests also passed with this alternative that's more similar to the original implementation, but it seemed too complicated.
```suggestion
    separator = typeof separator === 'undefined' ? ',' : separator;
    const first = this.length === 1 ? String(this.first) : String(this.first) + separator;
    return first + this.rest.join(separator);
```

Reply:
Ugh... you are right. It probably should be. Sheesh, I hate that `undefined` behavior of join... too late to make separator a required argument? 😆